inspirehep/modules/records/serializers/impactgraph_serializer.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Impact Graph serializer for records."""
+
+from __future__ import absolute_import, print_function
+
+import json
+
+from invenio_search import current_search_client
+from invenio_search.api import Query
+
+from inspirehep.utils.record import get_title
+
+
+class ImpactGraphSerializer(object):
+
+    """Impact Graph serializer for records."""
+
+    def serialize(self, pid, record, links_factory=None):
+        """
+        Serialize a single impact graph from a record.
+
+        :param pid: Persistent identifier instance.
+        :param record: Record instance.
+        :param links_factory: Factory function for the link generation,
+                              which are added to the response.
+        """
+        out = {}
+
+        # Add information about current record
+        out['inspire_id'] = record['control_number']
+        out['title'] = get_title(record)
+        out['year'] = record['earliest_date'].split('-')[0]
+
+        # Get citations
+        citations = []
+
+        es_query = Query('refersto:' + record['control_number'])
```

Comment:
Is this ES or `invenio_search`? I think the latter. Because otherwise in ES we could also filter out the fields without all the massive amount of metadata.